docs/book/advanced-guide/stacks/custom-flavors.md

```diff
@@ -220,12 +220,31 @@ class MyArtifactStoreFlavor(BaseArtifactStoreFlavor):
 
 ## Managing a Custom Stack Component Flavor
 
-Once your implementation is complete, you can register it through the CLI:
+Once your implementation is complete, you can register it through the CLI.
+Please ensure you **point to the flavor class via dot notation**:
 
 ```shell
 zenml artifact-store flavor register <path.to.MyArtifactStoreFlavor>
 ```
 
+For example, your flavor class `MyArtifactStoreFlavor` is defined in `flavors/my_flavor.py`,
+you'd register it by doing:
+
+```shell
+zenml artifact-store flavor register flavors.my_flavor.MyArtifactStoreFlavor
+```
+
+{% hint style="warning" %}
+ZenML resolves the flavor class by taking the path where you initialized zenml
+(via `zenml init`) as the starting point of resolution. Therefore, please ensure
+you follow [the best practice](../../guidelines/best-practices.md) of initializing
+zenml at the root of your respository.
+
+If ZenML does not find an initialized ZenML repository in any parent directory, it
+will default to the current working directory, but usually its better to not have to
```

Comment:
I think this is incorrect right? It doesn't use the current working directory, but the parent directory of the python main module (If running `python run.py`, these will be the same as the working dir. You could also do `python some_dir/run.py`, in which case they won't match)

Reply:
Actually in the case of the flavor command, even that is incorrect as the main module will be the ZenML CLI script.

Reply:
I'll have a call with Stefan later where will discuss the same issue for `zenml pipeline build/run`, I'll report back what we're going to do in this case.

Reply:
What we decided: Your doc update here regarding the working directory is the way we want this to work, but not the way it's implemented right now. So we'd also have to fix the code to actually make it work that way

Reply:
Continuing my conversation here: I'll be fixing this in a generic way on my branch, no need to do anything here

Reply:
@schustmi Should I then go ahead and merge this when its ready or will your branch not be ready before next release?

Reply:
Go for it